test/component-test.js

```diff
@@ -35,8 +35,17 @@ describe('component creation', () => {
     });
 
     context('node is scoped', () => {
-      it('applies text content to element');
-      it('recursively creates elements with scoped data');
+      it('applies text content to element', () => {
+        var attr = new Attribute('data-prop', 'lol');
+        var node = new Node('div', [attr]);
+        var data = {lol: 'wat'};
+
+        var {tag, props, content} = createComponent(node, data);
```

Comment:
It might just be my lack of familiarity with ES6, but I feel like the object destructuring here obscures the intent of the usage a tad. I don't feel like in normal production usage you'd be destructuring the result of the createComponent, right? So even though the destructuring gives you all the data you need for test expectations, I feel like there's some additional documentation benefit of _not_ destructuring, to show how it'd be more likely to be used.

For example, here: https://github.com/iamvery/ratchetjs/pull/4/commits/0f4fa6832879714f250589aba09f9ff634391093#diff-e5a50c1d49bc4ef25ffa3477ed0d7656R21

I like how there was a `first` and `last` node, and the `content` property was read off of each. Preserving the node objects as-is feels a bit more readable to me.